### ISSUE TITLE
Typescript support for External Source Plugins

### ISSUE THREAD

**Issue by lschierer** [ProjectEvergreen/greenwood#1296]
## Type of Change
New Feature

## Summary
Please support using TypeScript to write an External Source Plugin 

## Details

I attempted to follow https://www.greenwoodjs.io/plugins/source/ changing the .js extension out for .ts and get an ERR_UNKNOWN_FILE_EXTENSION that persists even after I was assisted in solving these errors for other sections of my configuration [here](https://github.com/ProjectEvergreen/greenwood/discussions/1292).   

I suspect that its actually possible to do this, but that it would be (currently) a multi-step process, where I'd need to transpile the plugin to javascript to then be able to run the greenwood develop or build commands.  This request is to support, and document, a configuration that will consolidate that into a single command line that could be included as the develop script in a package.json file.  

I further suspect that this is related to  [#1250](https://github.com/ProjectEvergreen/greenwood/issues/1250) where in the comments I requested support for the main configuration file to be supported as a typescript file.  My intuition is that if you solved that, this would be mostly solved as well.

**Comment by thescientist13** [comment]
> I further suspect that this is related to https://github.com/ProjectEvergreen/greenwood/issues/1250 where in the comments I requested support for the main configuration file to be supported as a typescript file. My intuition is that if you solved that, this would be mostly solved as well.

Yeah, pretty much.  Right now the TypeScript plugin is only geared towards transforming "web" code, e.g. anything in your _src/_ directory, but not really anything else.

Let me align this with #1250 though since I think this would be something we would apply to all plugins at large.  My guess is that this would likely try and tackle this in two phases:
1. Support exposing JS Docs style types (#1250)
1. Have Greenwood support TypeScript files first party for its APIs (would probably re-purpose this issue for that effort)

I've also been experimenting with Bun and Deno and they are working pretty well out of the box with some Greenwood demo apps as part of initial testing, and so now with NodeJS also supporting TS, I may even considering moving TS support right into Greenwood itself, probably with a flag though.  We shall see!
